pkg/vfile/vfile.go

```diff
@@ -214,3 +249,37 @@ func openHashedFile(path string, wantHash []byte, h hash.Hash) (*File, error) {
 	}
 	return f, nil
 }
+
+func CheckHashedContent(b *bytes.Reader, wantHash []byte, h hash.Hash) (*bytes.Reader, error) {
+	if len(wantHash) == 0 {
+		return b, ErrInvalidHash{
+			Err: ErrNoExpectedHash,
+		}
+	}
+
+	got, err := CalculateHash(b, h)
+	if err != nil {
+		return b, err
+	}
+
+	if !bytes.Equal(wantHash, got) {
```

Comment:
might be overkill, but also you wouldn't have to think about it: consider [subtle](https://pkg.go.dev/crypto/subtle).ConstantTimeCompare

Reply:
Interesting. Updated to the subtle function.
Squashed.